include/zephyr/bluetooth/buf.h

```diff
@@ -97,13 +97,13 @@ struct bt_buf_data {
   * available for the HCI driver to allocate from.
   *
   * TODO: When CONFIG_BT_BUF_ACL_RX_COUNT is removed,
-  *       remove the MAX and only keep (CONFIG_BT_MAX_CONN + 1)
+  *       remove the MAX and only keep the 1.
   */
-#define BT_BUF_ACL_RX_COUNT                                                                        \
-	(MAX(CONFIG_BT_BUF_ACL_RX_COUNT, (CONFIG_BT_MAX_CONN + 1)) +                               \
-	 CONFIG_BT_BUF_ACL_RX_COUNT_EXTRA)
+#define BT_BUF_ACL_RX_COUNT_EXTRA CONFIG_BT_BUF_ACL_RX_COUNT_EXTRA
+#define BT_BUF_ACL_RX_COUNT       (MAX(CONFIG_BT_BUF_ACL_RX_COUNT, 1) + BT_BUF_ACL_RX_COUNT_EXTRA)
```

Comment:
Still thinking on this, `BT_BUF_HCI_ACL_RX_COUNT` introduced in this PR is ending up being `1` always! 

Thinking aloud,
`BT_BUF_HCI_ACL_RX_COUNT` should able to be greater than or equal to 1, so that more number of ACL buffers be buffered up for host processing (in bursts, x bps) to match the reception over transport (in bursts, y bps).

At the same time, `BT_BUF_ACL_RX_COUNT_EXTRA` be for L2CAP recombination, so that `BT_BUF_HCI_ACL_RX_COUNT`  is available for ACL buffer with and without Controller to Host data flow control.

Reply:
Created for follow up: https://github.com/zephyrproject-rtos/zephyr/issues/86142